contracts/cw-proposal-multiple/src/query.rs

```diff
@@ -0,0 +1,132 @@
+use crate::{
+    proposal::MultipleChoiceProposal,
+    state::{Config, VoteInfo},
+};
+use cosmwasm_std::{Decimal, Uint128};
+use cw3::Status;
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use voting::voting::MultipleChoiceVotes;
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+pub struct ProposalListResponse {
+    pub proposals: Vec<ProposalResponse>,
+}
+
+/// Information about a proposal returned by proposal queries.
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+pub struct ProposalResponse {
+    pub id: u64,
+    pub proposal: MultipleChoiceProposal,
+}
+
+/// Information about the current status of a proposal.
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+pub struct VoteTallyResponse {
+    /// Current proposal status
+    pub status: Status,
+    /// Required passing criteria
+    pub threshold: ThresholdResponse,
+    /// Current percentage turnout
+    pub quorum: Decimal,
+    /// Total number of votes per option for the proposal
+    pub total_votes: Vec<Uint128>,
+    /// Total number of votes possible for the proposal
+    pub total_weight: Uint128,
+    /// Tally of the different votes
+    pub votes: MultipleChoiceVotes,
+}
+
+/// This defines the different ways tallies can happen.
+/// Every contract should support a subset of these, ideally all.
+///
+/// The total_weight used for calculating success as well as the weights of each
+/// individual voter used in tallying should be snapshotted at the beginning of
+/// the block at which the proposal starts (this is likely the responsibility of a
+/// correct cw4 implementation).
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum ThresholdResponse {
```

Comment:
Seems like we only support quorum here. Any reason to have this entire enum?

Reply:
artifact of older version, removed